src/Microsoft.Management.Configuration.Processor/DSCv3/Helpers/ProcessExecution.cs

```diff
@@ -168,6 +174,68 @@ public ProcessExecution Start()
                 startInfo.RedirectStandardInput = true;
             }
 
+            foreach (var env in this.EnvironmentVariables)
+            {
+                switch (env.ValueType)
+                {
+                    case ProcessExecutionEnvironmentVariableValueType.Override:
+                        startInfo.EnvironmentVariables[env.Name] = env.Value;
+                        break;
+
+                    case ProcessExecutionEnvironmentVariableValueType.Prepend:
+                        {
+                            string existingValue = startInfo.EnvironmentVariables[env.Name] ?? string.Empty;
+                            if (string.IsNullOrEmpty(env.Separator))
+                            {
+                                startInfo.EnvironmentVariables[env.Name] = env.Value + existingValue;
+                            }
+                            else
+                            {
+                                if (existingValue.StartsWith(env.Separator) && env.Value.EndsWith(env.Separator))
+                                {
+                                    startInfo.EnvironmentVariables[env.Name] = env.Value.Substring(0, env.Value.Length - env.Separator.Length) + existingValue;
+                                }
+                                else if (existingValue.StartsWith(env.Separator) || env.Value.EndsWith(env.Separator))
+                                {
+                                    startInfo.EnvironmentVariables[env.Name] = env.Value + existingValue;
+                                }
+                                else
+                                {
+                                    startInfo.EnvironmentVariables[env.Name] = env.Value + env.Separator + existingValue;
+                                }
+                            }
+
+                            break;
+                        }
+
+                    case ProcessExecutionEnvironmentVariableValueType.Append:
+                        {
+                            string existingValue = startInfo.EnvironmentVariables[env.Name] ?? string.Empty;

```

Comment:
Code duplicated; could refactor out and just swap which value was first and second.